microcosm_flask/conventions/upload.py

```diff
@@ -0,0 +1,44 @@
+"""
+Conventions for file upload.
+
+"""
+from flask import request
+from microcosm_flask.conventions.base import Convention
+from microcosm_flask.operations import Operation
+from werkzeug.exceptions import BadRequest
+
+
+class UploadConvention(Convention):
+
+    def configure_upload(self, ns, definition):
+        """
+        Register an upload endpoint.
+
+        The definition's func should be an upload function, which must:
+        - accept kwargs for path data
+        - accept a dict of files
+
+        :param ns: the namespace
+        :param definition: the endpoint definition
+
+        """
+        @self.graph.route(ns.collection_path, Operation.Upload, ns)
+        def upload(**path_data):
+            files = request.files
+
+            if not files:
+                raise BadRequest("No files uploaded!")
+
+            definition.func(files, **path_data)
```

Comment:
It looks like above a certain fixed threshold (500Kb?) Flask will not make the file data available for streaming in memory from `file.read()` without saving it to temporary location first - http://stackoverflow.com/a/42405115
might be fine for most use cases to assume this size but in case does go over, the method linked to above uses flask's recommended secure_filename for handling saving it